### PR TITLE
Support access log to print traceID and connectionID

### PR DESCRIPTION
**Pull request by Bryce-huang**
### Issues associated with this PR

https://github.com/mosn/mosn/issues/2105

### Solutions

add accesslog format && trace and  conn id variable

### UT result
Unit Test is needed if the code is changed, your unit test should cover boundary cases, corner cases, and some exceptional cases. And you need to show the UT result.

### Benchmark
If your code involves the processing of every request, you should give the Benchmark Result.

### Code Style
+ Make sure `Goimports` has run
+ Show `Golint` result

### CODE REVIEW

**Comment by Bryce-huang** [comment]
pls review this pr

File: pkg/proxy/var.go [hunk 1]
```diff
@@ -268,3 +271,21 @@ func responseHeaderMapGetter(ctx context.Context, value *variable.IndexedValue,
...
+	return variable.ValueNotFound, errors.New("not found connection id")
+}
+
+// TraceIDGetter
+// get request's trace ID
+func traceIDGetter(ctx context.Context, value *variable.IndexedValue, data interface{}) (string, error) {
+	if id := mosnctx.Get(ctx, types.ContextKeyTraceId); id != "" {
+		return id.(string), nil
+	}
+	return variable.ValueNotFound, errors.New("not found connection id")
```
Comment (taoyuanyuan) [inline pkg/proxy/var.go:290]: connection -> traceID

File: pkg/proxy/var.go [hunk 2]
```diff
@@ -268,3 +271,21 @@ func responseHeaderMapGetter(ctx context.Context, value *variable.IndexedValue,
 
 	return string(headerValue), nil
 }
+
+// ConnectionIDGetter
+// get request's connection ID
+func connectionIDGetter(ctx context.Context, value *variable.IndexedValue, data interface{}) (string, error) {
+	if id := mosnctx.Get(ctx, types.ContextKeyConnectionID); id != "" {
```
Comment (taoyuanyuan) [inline pkg/proxy/var.go:278]: pls support `ContextUpstreamConnectionID` also.

File: pkg/proxy/var.go [hunk 3]
```diff
@@ -268,3 +271,21 @@ func responseHeaderMapGetter(ctx context.Context, value *variable.IndexedValue,
...
+		return strconv.FormatUint(id.(uint64), 10), nil
```
Comment (taoyuanyuan) [inline pkg/proxy/var.go:279]: judge whether `id` is uint64 or not.

File: pkg/types/accesslog.go [hunk 4]
```diff
@@ -18,6 +18,6 @@
 package types
 
 // DefaultAccessLogFormat provides a pre-defined format
-const DefaultAccessLogFormat = "%start_time% %request_received_duration% %response_received_duration% %bytes_sent%" + " " +
+const DefaultAccessLogFormat = "%start_time% %trace_id%  %connection_id% %context_key_connection_id% %request_received_duration% %response_received_duration% %bytes_sent%" + " " +
```
Comment (taoyuanyuan) [inline pkg/types/accesslog.go:21]: rename to `upstream_connection_id`.

File: pkg/types/variable.go [hunk 5]
```diff
@@ -39,6 +39,9 @@ const (
 	VarRequestedServerName            string = "requested_server_name"
 	VarRouteName                      string = "route_name"
 	VarProtocolConfig                 string = "protocol_config"
+	VarConnectionID                   string = "connection_id"
+	VarTraceID                        string = "trace_id"
+	VarContextKeyConnectionID         string = "context_key_connection_id"
```
Comment (taoyuanyuan) [inline pkg/types/variable.go:44]: rename to `VarUpstreamConnectionID`

File: pkg/proxy/var.go [hunk 6]
```diff
@@ -268,3 +272,34 @@ func responseHeaderMapGetter(ctx context.Context, value *variable.IndexedValue,
...
+func traceIDGetter(ctx context.Context, value *variable.IndexedValue, data interface{}) (string, error) {
+	if id := mosnctx.Get(ctx, types.ContextKeyTraceId); id != "" {
+		return id.(string), nil
+	}
+	return variable.ValueNotFound, errors.New("not found traceID")
+}
+
+// ContextUpstreamConnectionID
+// get request's ContextUpstreamConnectionID
+func contextUpstreamConnectionIDGetter(ctx context.Context, value *variable.IndexedValue, data interface{}) (string, error) {
```
Comment (taoyuanyuan) [inline pkg/proxy/var.go:298]: upstreamConnectionIDGetter

**Review by taoyuanyuan — APPROVED** [review]
LGTM

File: pkg/proxy/var.go [hunk 7]
```diff
@@ -268,3 +272,34 @@ func responseHeaderMapGetter(ctx context.Context, value *variable.IndexedValue,
...
+// get request's trace ID
+func traceIDGetter(ctx context.Context, value *variable.IndexedValue, data interface{}) (string, error) {
+	if id := mosnctx.Get(ctx, types.ContextKeyTraceId); id != "" {
+		return id.(string), nil
+	}
+	return variable.ValueNotFound, errors.New("not found traceID")
+}
+
+// UpstreamConnectionIDGetter
+// get request's upstreamConnectionIDGetter
```
Comment (taoyuanyuan) [inline pkg/proxy/var.go:297]: upstreamConnectionIDGetter -> upstream connection ID

**Review by taoyuanyuan — APPROVED** [review]
LGTM

**Review by doujiang24 — APPROVED** [review]

**Pull request merged**